### ISSUE TITLE
Error when retrieving OSV

### ISSUE THREAD

**Issue by dctrwatson** [mend/renovate-ce-ee#446]
Starting in with v6.10.0, renovate now errors out when it tries to update OSV. So features like the dep dashboard and auto-closing PRs no longer work.

```
      "err": {
        "name": "HttpError",
        "status": 401,
        "response": {
          "url": "https://api.github.com/repos/renovatebot/osv-offline/releases",
          "status": 401,
          "headers": {
            "access-control-allow-origin": "*",
            "access-control-expose-headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
            "connection": "close",
            "content-length": "80",
            "content-security-policy": "default-src none",
            "content-type": "application/json; charset=utf-8",
            "date": "Fri, 02 Feb 2024 01:54:53 GMT",
            "referrer-policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
            "server": "GitHub.com",
            "strict-transport-security": "max-age=31536000; includeSubdomains; preload",
            "vary": "Accept-Encoding, Accept, X-Requested-With",
            "x-content-type-options": "nosniff",
            "x-frame-options": "deny",
            "x-github-media-type": "github.v3; format=json",
            "x-github-request-id": "1EFC:46B3:71B5CF:E768FB:65BC4B6D",
            "x-ratelimit-limit": "60",
            "x-ratelimit-remaining": "22",
            "x-ratelimit-reset": "1706839564",
            "x-ratelimit-resource": "core",
            "x-ratelimit-used": "38",
            "x-xss-protection": "0"
          },
          "data": {
            "message": "Bad credentials",
            "documentation_url": "https://docs.github.com/rest"
          }
        },
        "request": {
          "method": "GET",
          "url": "https://api.github.com/repos/renovatebot/osv-offline/releases",
          "headers": {
            "accept": "application/vnd.github.v3+json",
            "user-agent": "octokit-rest.js/20.0.2 octokit-core.js/5.1.0 Node.js/18.19.0 (linux; x64)",
            "authorization": "***********"
          },
          "request": {
            "fetch": "[function]",
            "hook": "[function]"
          }
        },
        "message": "Bad credentials",
        "stack": "HttpError: Bad credentials\n    at /usr/src/app/node_modules/renovate/node_modules/@octokit/request/dist-node/index.js:124:21\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at tryDownloadDb (/usr/src/app/node_modules/renovate/node_modules/@renovatebot/osv-offline/dist/lib/download.js:46:26)\n    at OsvOffline.initialize (/usr/src/app/node_modules/renovate/node_modules/@renovatebot/osv-offline/dist/lib/osv-offline.js:14:24)\n    at Function.create (/usr/src/app/node_modules/renovate/node_modules/@renovatebot/osv-offline/dist/lib/osv-offline.js:26:9)\n    at Vulnerabilities.initialize (/usr/src/app/node_modules/renovate/lib/workers/repository/process/vulnerabilities.ts:56:23)\n    at Function.create (/usr/src/app/node_modules/renovate/lib/workers/repository/process/vulnerabilities.ts:61:5)\n    at getDashboardMarkdownVulnerabilities (/usr/src/app/node_modules/renovate/lib/workers/repository/dependency-dashboard.ts:502:32)\n    at ensureDependencyDashboard (/usr/src/app/node_modules/renovate/lib/workers/repository/dependency-dashboard.ts:415:16)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:92:9)\n    at attributes.repository (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:184:11)\n    at start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:169:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:18:22"
      },
      "msg": "Repository has unknown error"
    }
```

I assume it's related to https://github.com/renovatebot/osv-offline/commit/d84d403f8a9a8b52639d6a3000b5e2633ebf845d

Rolling back to v6.9.1 solves the issue.

**Comment by nabeelsaabna** [comment]
Hi @dctrwatson 
Renovate CE/EE uses [Renovate](https://github.com/renovatebot/renovate) as is.
- `renovate-ce:6.9.1` uses `renovate@37.137.1`
- `renovate-ce:6.10.0` uses `renovate@37.153.2`

Please open a new issue in [osv-offline/issues](https://github.com/renovatebot/osv-offline/issues) for fixing/correcting the behavior, fix will be available in renovate-ce/ee only after its part of `renovatebot/osv-offline` and/or `renovatebot/renovate`

**Comment by dctrwatson** [comment]
@nabeelsaabna I believe this comes from how renovate-ce overrides the `GITHUB_COM_TOKEN` when using the `MEND_RNV_GITHUB_APP_*` vars. I tried setting `GITHUB_COM_TOKEN` with a valid token but that did not fix the issue.

**Comment by nabeelsaabna** [comment]
renovate-ce does nothing with/to the `GITHUB_COM_TOKEN`
can you please check the log message `"msg": "Running Renovate job"`
it should be something like:
```
{
    "name": "renovate",
    "level": 20,
    "logContext": "stHRVJ8METYH3J4w1M6UG",
    "job": {
        "jobId": "5ce72c1b-119a-4938-ac16-80a6135190f9",
        "reason": "hourly",
        "priority": 10,
        "addedAt": "2024-02-06 02:00:00.347199",
        "startedAt": "2024-02-06 02:00:10.924213",
        "lastAlive": "2024-02-06 02:00:10.924213",
        "attempts": 1,
        "repository": "<REPO>",
        "organizationName": "<ORG>",
        "installationId": 1234
    },
    "config": {
        "dependencyDashboardFooter": "\n- [ ] <!-- manual job -->Check this box to trigger a request for Renovate to run again on this repository\n",
        "endpoint": "https://github.test.com/api/v3/",
        "gitAuthor": "test-r[bot]@users.noreply.github.test.com",
        "logContext": "d5eacf43de",
        "platform": "github",
        "repositories": [
            "<REPO>"
        ],
        "token": "***********",
        "username": "test-r[bot]"
    },
    "msg": "Running Renovate job",
    "time": "2024-02-06T02:00:10.988Z",
    "v": 0
}
```

the `config` object is simply passed to Renovate-CLI via `RENOVATE_CONFIG` environment variable

**Comment by dctrwatson** [comment]
Yup, the config contains a `token` even though I do not have `GITHUB_COM_TOKEN` set

**Comment by nabeelsaabna** [comment]
@dctrwatson this means it's an different behavior from [Renovate](https://github.com/renovatebot/renovate) and I recommend opening an issue in that repository
